src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClient.java

```diff
@@ -75,7 +75,7 @@ public CourtCaseRestClient(@Qualifier("courtCaseServiceWebClient") WebClient web
                 return courtCaseResponse;
             })
             .onErrorResume((e) -> {
-                log.debug("GET failed for retrieving the case for path {}", path, e);
+                // This is normal in the context of CCM, don't log
```

Comment:
Is this to fix the exceptions for case not found?  If so I think it will still register the CourtCaseNotFoundException in App insights as it is looking for `exception` data types rather than `trace` (log messages). Is it line 198 which needs changing to stop using an exception altogether?

Reply:
I think you're probably right, but one issue we might find with changing line 198 is that the reliance on Monos introduces a form of coupling which can cause changes to cascade beyond the client e.g. If the RestClient decides that an error is now an empty - the caller then has to change its behaviour. I've found WebClient is a little unforgiving in that respect and it's easy to trip up if you don't get the error handling right first time. 

Can I suggest we put some time aside to review together how we're using WebClient?  I think that there's room for improvement in terms of writing 'idiomatic' WebClient code and we've learned a lot about it since we started so this would be a good opportunity to turn that into some best-practices and start applying them consistently.  It would be good to abstract away as much of the standard error handling as possible as that's where I tend to trip up but it will probably require a bit of dedicated effort to get it right.